File: code/sorting/selection_sort/selection_sort.kt
// Part of Cosmos by OpenGenus Foundation

import java.util.*

fun <T : Comparable<T>> sort(array: Array<T>) {
    for (i in array.indices) {
        var min = i
        for (j in i + 1 until array.size) {
            /* find local min */
            if (array[j] < array[min]) {
                min = j
            }
        }
        val temp = array[i]
        array[i] = array[min]
        array[min] = temp
    }
}

fun main(args: Array<String>) {
    val arr = arrayOf(1, 5, 2, 5, 2, 9, 7)
    sort(arr)
    print(Arrays.toString(arr))
}

File: cosmos-master/code/sorting/selection_sort/SelectionSort.php
<?php
// Part of Cosmos by OpenGenus Foundation
function selection_sort($arr) {
    $n = count($arr);

    foreach ($arr as $k => $el) {
        $min = $k;

        for ($j = $k + 1; $j < $n; $j++) {
            if ($arr[$j] < $arr[$min]) {
                $min = $j;
            }
        }

        $hold = $el;
        $el = $arr[$min];
        $arr[$min] = $hold;
    }

    return $arr;
}


$test = [1, 3, 43, 2, 14, 53, 12, 87, 33];

print_r($test);
$sorted = selection_sort($test);

print_r($sorted);
